### ISSUE TITLE
Joypad buttons not working ?

### ISSUE THREAD

**Issue by blinkydoos** [midwan/amiberry#172]
Hi, I built Amiberry Dev SDL1 yesterday but it seems that none of the buttons on the joypad seem to work, the axis/d-pad works. Not sure if it's a problem with Amiberry, is it still reading the Controller configs from RetroArch or has that been removed ? Can anyone else confirm this ?

**Comment by CypherXG** [comment]
Which Joypad?
Config in the correct Path?
Path correct in the Emulator?

**Comment by HoraceAndTheSpider** [comment]
You can tell if your controller has been picked up with a RetroArch cfg file by going to the custom panel, selecting the appropriate input and looking for the [R] adjacent to the device name.

**Comment by HoraceAndTheSpider** [comment]
http://www.ultimateamiga.co.uk/HostedProjects/RetroPieAmiga/guide/Amiberry_Menu13_CustomPanel.png

**Comment by blinkydoos** [comment]
Yes, my controller is recognized with [R] in Custom Controls but only Up,Down,Left & Right work ingame. I have my controller config file in `emulators/amiberry/controllers`.

The RetroArch path seems to have changed to `amiberry/conf/` instead of `retropie/configs/all/retroarch/autoconfig/`

I changed the path back to the original RetroArch one but still doesn't work properly.

It worked fine in a previous DEV SDL1 build. I must be the only person using the latest version with a controller ! :)

**Comment by HoraceAndTheSpider** [comment]
Well I’ve just been testing multiple controllers with the dev branch, all fine. Build is about 36 hours old.

So if your controller Config file is in `emulators/amiberry/controllers` then you need to set the controller path to that, and not the retroarch one, after that maybe try reloading Amiberry. The [R] is the key point though and if that is showing there should be no reason it is not working, although perhaps check what joystick mode you have set?

(The behaviour you describe sounds lie what happens when the port is in “mouse” mode)

All it is doing is checking that location for a text file... nothing magic there at all. 


I will check with @midwan of there have been any nightly changes that might affect normal button presses.

**Comment by blinkydoos** [comment]
All my paths are the same as always, as I said, it works fine on a previous build. I tried changing the Controller mode to CD32 and the buttons work but if I change it back to Joystick they stop working and just the D-Pad works.

**Comment by HoraceAndTheSpider** [comment]
Is it implicitly in “joystick” mode the y do not work, or in “default” mode do they work/not work also? 

Could be related to the changes which should only affect keyboard controllers then.

**Comment by blinkydoos** [comment]
'Default' mode also doesn't work, only D-Pad.

**Comment by midwan** [comment]
Confirmed bug, something must have got broken recently. Will find it and fix it :)

**Comment by midwan** [comment]
This is caused by 809781267dd3b126e11d9089766986146f92c91e

**Comment by blinkydoos** [comment]
Thanks, i'll try it tomorrow.
